qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -121,26 +125,141 @@ def _truncate_inactive_qubits(
+    synthesis_method: str = DEFAULT_SYNTHESIS_METHOD,
+) -> QuantumCircuit:
+    """Synthesize a circuit of a Clifford element. The resulting circuit contains only
+    ``basis_gates`` and it complies with ``coupling_tuple``.
+
+    Args:
+        clifford: Clifford element to be converted
+        basis_gates: basis gates to use in the conversion
+        coupling_tuple: coupling map to use in the conversion in the form of tuple of edges
+        synthesis_method: conversion algorithm name
+
+    Returns:
+        Synthesized circuit
+    """
+    qc = QuantumCircuit(clifford.num_qubits, name=str(clifford))
+    qc.append(clifford, qc.qubits)
+    return _synthesize_clifford_circuit(
+        qc,
+        basis_gates=basis_gates,
+        coupling_tuple=coupling_tuple,
+        synthesis_method=synthesis_method,
+    )
+
+
 def _synthesize_clifford_circuit(
-    circuit: QuantumCircuit, basis_gates: Tuple[str]
+    circuit: QuantumCircuit,
+    basis_gates: Optional[Tuple[str]],
+    coupling_tuple: Optional[Tuple[Tuple[int, int]]] = None,
+    synthesis_method: str = DEFAULT_SYNTHESIS_METHOD,
 ) -> QuantumCircuit:
-    # synthesizes clifford circuits using given basis gates, for use during
-    # custom transpilation during RB circuit generation.
-    return transpile(circuit, basis_gates=list(basis_gates), optimization_level=1)
+    """Convert a Clifford circuit into one composed of ``basis_gates`` with
+    satisfying ``coupling_tuple`` using the specified synthesis method.
+
+    Args:
+        circuit: Clifford circuit to be converted
+        basis_gates: basis gates to use in the conversion
+        coupling_tuple: coupling map to use in the conversion in the form of tuple of edges
+        synthesis_method: name of Clifford synthesis algorithm to use
+
+    Returns:
+        Synthesized circuit
+    """
+    if basis_gates:
+        basis_gates = list(basis_gates)
+    coupling_map = CouplingMap(coupling_tuple) if coupling_tuple else None
+
+    # special handling for 1q or 2q case for speed
+    if circuit.num_qubits <= 2:
+        if synthesis_method == DEFAULT_SYNTHESIS_METHOD:
+            return transpile(
+                circuit,
+                basis_gates=basis_gates,
+                coupling_map=coupling_map,
+                optimization_level=1,
+            )
+        else:
+            # Provided custom synthesis method, re-synthesize Clifford circuit
+            # convert the circuit back to a Clifford object and then call the synthesis plugin
+            new_circuit = QuantumCircuit(circuit.num_qubits, name=circuit.name)
+            new_circuit.append(Clifford(circuit), new_circuit.qubits)
+            circuit = new_circuit
 
+    # for 3q+ or custom synthesis method, synthesizes clifford circuit
+    hls_config = HLSConfig(clifford=[(synthesis_method, {"basis_gates": basis_gates})])
+    pm = PassManager([HighLevelSynthesis(hls_config=hls_config, coupling_map=coupling_map)])
+    circuit = pm.run(circuit)
```

Comment:
Thank you for providing this useful HighLevelSynthesisPlugin interface. I like it and it's sufficient for this PR.
Let me share a minor point that could be simplified. I was a bit confused with the fact that `HighLevelSynthesisPlugin.run` takes over `coupling_map` option set up via `HighLevelSynthesis.__init__`, but it does not take over `basis_gates` option set up via `HighLevelSynthesis.__init__`, so I had to set it via `hls_config` [here](https://github.com/Qiskit-Extensions/qiskit-experiments/blob/33dddf8049fb6a7960fd7c967e5f1f641615b2fd/qiskit_experiments/library/randomized_benchmarking/clifford_utils.py#L195). Is this (`run` does not have `basis_gates` option) intentional?
It was not clear to me which options should be given through `hls_config` (`HLSConfig`) and which options should be given directly (like `coupling_map`) in the current API. Probably, more explanation on that point in [the API reference doc](https://docs.quantum.ibm.com/api/qiskit/qiskit.transpiler.passes.HighLevelSynthesis) would be helpful for developers like me.

As for an extension, it may be worth considering an introduction of `HighLevelSynthesis.synthesize` for the ease of custom Clifford-to-QuantumCircuit conversion as I mentioned [above](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1288#discussion_r1454461486).